app/javascript/components/App.js

```diff
@@ -44,7 +67,18 @@ class App extends Component {
             <Route path="/apartmentindex" render = {(props) => <ApartmentIndex {...props} apartments={this.state.apartments} />} />
             <Route path="/apartmentshow" component={ApartmentShow} />
             <Route path="/apartmentnew" component={ApartmentNew} />
-            <Route path="/apartmentedit" component={ApartmentEdit} />
+            {/* <Route path="/apartmentedit/:id" render = {(props) => {
+              const { match } = props
+              const { id } = match.params
+              return <ApartmentEdit {...props} apartment={this.state.apartments.find(apartment => apartment.id === id)} />
+            }
+            } /> */}
+            <Route path="/apartmentedit/:id" render={(props) => {
+              let id = props.match.params.id
+              let apartment = this.state.apartments.find(apartment => apartment.id === id)
```

Comment:
Both of these approaches are fine, however props.match.params.id will come back as a string not a number, so evaluating it as a number will throw errors.  Also, please remove the commented-out code.

Reply:
props.match.params.id will come back as a string not a number, so evaluating it as a number will throw errors.  Please make sure to convert id to a number.